File: buerguer-pig/AMarotoChivite-BurguerPig/AMarotoChivite-BurguerPig-Java/src/main/java/Ficheros/BurguerPig/mappers/base/ListMapper.java
package Ficheros.BurguerPig.mappers.base;

import java.util.List;
import java.util.stream.Collectors;

public class ListMapper<Dto, Model> implements IDtoMapper<ListDto<Dto>, List<Model>> {

    private IDtoMapper<Dto, Model> objectMapper;

    public ListMapper(IDtoMapper<Dto, Model> objectMapper) {
        this.objectMapper = objectMapper;
    }

    @Override
    public ListDto<Dto> toDto(List<Model> model) {
        List<Dto> dtoList = model.stream().map(objectMapper::toDto).collect(Collectors.toList());
        return new ListDto<>(dtoList);
    }

    @Override
    public List<Model> toModel(ListDto<Dto> dto) {
        return dto.list.stream().map(objectMapper::toModel).collect(Collectors.toList());
    }
}

File: buerguer-pig/AMarotoChivite-BurguerPig/AMarotoChivite-BurguerPig-Kotlin/src/main/kotlin/Ficheros/BurguerPig/mappers/base/ListMapper.kt
package Ficheros.BurguerPig.mappers.base

class ListDto<Dto>(val list: List<Dto>)

open class ListMapper<Dto, Model>(private val objectMapper: IDtoMapper<Dto, Model>) :
    IDtoMapper<ListDto<Dto>, List<Model>> {

    override fun toDto(model: List<Model>): ListDto<Dto> {
        val dtoList = model.map { objectMapper.toDto(it) }
        return ListDto(dtoList)
    }

    override fun toModel(dto: ListDto<Dto>): List<Model> {
        return dto.list.map { objectMapper.toModel(it) }
    }
}